index.js

```diff
@@ -3,191 +3,147 @@ const express=require('express');
-    let user=users.find(user=>user.username===req.params.username);
+//Create user
+app.post('/users',(req,res)=>{
+  Users.findOne({Username:req.body.Username})
+  .then(user=>{
     if(user){
-      user.password=req.params.password
-      res.status(201).send( user.username +'\'s passowrd has been succesfully updated')
+      res.status(400).send(req.body.Username + ' already exists!')
     }else{
-      res.status(400).send(req.params.username +' not found')
+      Users
+      .create({
+        Username: req.body.Username,
+        Password: req.body.Password,
+        Email: req.body.Email,
+        Birthday: req.body.Birthday
+      })
+      .then(user=> res.status(201).json(user))
+      .catch(error=>{
+        console.error(error);
+        res.status(500).send('Error: ' + error)
+      })
     }
- });
+  })
+  .catch(error=>{
+    console.error(error);
+    res.status(500).send('Error: ' + error)
+  })
+});
 
- //Allow users to add movies to their list of favourite movies
-app.put('/users/:username/favMovies/:favMovies',(req,res)=>{
-    let user=users.find(user=>user.username===req.params.username);
-    if(user){
-      const message=user.username +' succesfully added ' + req.params.favMovies + ' to his list of favourite movies ' 
-      user.favMovies.push(req.params.favMovies)
-      res.status(201).send(message)
-    }else{
-      res.status(400).send('The user ' + req.params.username +' is not found')
+//Update user info by username
+app.put('/users/:username', (req,res)=>{
+  Users.findOneAndUpdate({Username:req.params.username
+  },{ $set:
+    {
+      Username: req.body.Username,
+      Password: req.body.Password,
+      Email: req.body.Email,
+      Birthday: req.body.Birthday
     }
-  });
+  },
+  {new:true}
+  )
+  .then(updatedUser=>{
+    res.status(201).json(updatedUser)
+  })
+  .catch(error=>{
+    console.error(error);
+    res.status(500).send('Error :' + error)
+  })
+});
 
-  //Allow users to remove a movie from their list of favourite movies
-app.delete('/users/:username/favMovies/:removeMovie',(req,res)=>{
-    const {removeMovie}=req.params;
-    const {username}=req.params;
-  
-    let user=users.find(user=>user.username===username && user.favMovies.includes(removeMovie));
-    if(user){
-      const movieIndex=user.favMovies.indexOf(removeMovie)
-      const message= removeMovie + ' has been succesfully deleted from ' + username + '\'s list of favourite movies';
-      user.favMovies.splice(movieIndex,1)
-      res.status(200).send(message)
-    }else{
-      res.status(400).send( 'usrname :' + username + ' or movie:' +  removeMovie +' is not found')
-    }
-  
+
+ //Add new movie to list of favourite movies
+app.put('/users/:username/movies/:MovieID',(req,res)=>{
+   Users.findOneAndUpdate({Username:req.params.username
+  },
+  {$push:{FavouritesMovies:req.params.MovieID}},
+  {new:true})
+  .then(newMovie=>res.json(newMovie))
+  .catch(error=>{
+    console.error(error);
```

Comment:
most of the time the final code doesn't need console log, instead you can check out how to log you're system. using libraries like wastoni npm and bunyan npm